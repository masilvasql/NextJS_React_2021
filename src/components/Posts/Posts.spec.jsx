import {Posts   } from '.'
const { render, screen } = require("@testing-library/react")

const props = {
    posts:[
      {
        id:1,
        title:'title',
        body:'body',
        cover:'cover'
      },
      {
        id:2,
        title:'title 2',
        body:'body 2',
        cover:'cover 2'
      },
      {
        id:3,
        title:'title 3',
        body:'body 3',
        cover:'cover 3'
      },
    ]
  }

describe('<Posts/>', ()=>{
    it('should render posts', ()=>{
        render(<Posts {...props}/>);

        expect(screen.getAllByRole('heading', {name:/title/i}))
        .toHaveLength(3);

        expect(screen.getAllByRole('img', {name:/title/i}))
        .toHaveLength(3);

        expect(screen.getAllByText(/body/i))
        .toHaveLength(3);

        expect(screen.getByRole('img', {name:/title 3/i}))
        .toHaveAttribute('src', 'cover 3');
    })
    
    it('should match snapshot', ()=>{
        const {container } = render(<Posts {...props}/>);

        expect(container.firstChild).toMatchSnapshot();
    })

        
})