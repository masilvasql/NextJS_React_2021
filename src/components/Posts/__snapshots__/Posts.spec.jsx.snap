// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts/> should match snapshot 1`] = `
<div
  class="posts"
>
  <div
    class="post"
  >
    <img
      alt="title"
      src="cover"
    />
    <div
      class="post-content"
    >
      <h1>
        title
      </h1>
      <p>
        body
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="title 2"
      src="cover 2"
    />
    <div
      class="post-content"
    >
      <h1>
        title 2
      </h1>
      <p>
        body 2
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="title 3"
      src="cover 3"
    />
    <div
      class="post-content"
    >
      <h1>
        title 3
      </h1>
      <p>
        body 3
      </p>
    </div>
  </div>
</div>
`;
